rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -387,6 +396,153 @@ def _reorder_events(
         )
         return decoded_events
 
+    def _decode_staking_deposit_withdraw(
+            self,
+            context: DecoderContext,
+            expected_shares_event_type: HistoryEventType,
+            shares_event_type: HistoryEventType,
+            shares_event_subtype: HistoryEventSubType,
+            shares_notes_template: str,
+            expected_assets_event_type: HistoryEventType,
+            assets_event_type: HistoryEventType,
+            assets_event_subtype: HistoryEventSubType,
+            assets_notes_template: str,
+            is_deposit: bool,
+    ) -> EvmDecodingOutput:
+        """Decode yearn staking deposit and withdraw events, updating the events involving
+         the assets and shares of the gauge as specified by the parameters.
+         """
+        if (gauge_token := self.base.get_evm_token(address=context.tx_log.address)) is None:
+            log.error(f'Failed to find gauge token for yearn staking withdrawal {context.transaction}')  # noqa: E501
+            return DEFAULT_EVM_DECODING_OUTPUT
+
+        if (
+            gauge_token.underlying_tokens is None or
+            len(gauge_token.underlying_tokens) == 0 or
+            (vault_token := self.base.get_evm_token(address=gauge_token.underlying_tokens[0].address)) is None  # noqa: E501
+        ):
+            log.error(
+                f'Failed to find underlying vault token for yearn staking gauge token '
+                f'{gauge_token} in staking withdrawal {context.transaction}',
+            )
+            return DEFAULT_EVM_DECODING_OUTPUT
+
+        assets_amount = token_normalized_value(
+            token_amount=int.from_bytes(context.tx_log.data[0:32]),
+            token=vault_token,
+        )
+        shares_amount = token_normalized_value(
```

Comment:
hmm this makes no sense. You are taking the same part ofthe data for `assets_amount` and for `shares_amount`.

How come this works?